content/usgs-packages/geoknife_Discovery.Rmd

```diff
@@ -0,0 +1,233 @@
+  sprintf("<img src='../%s%s-%d.%s'/ title='%s'/>", 
+          options$fig.path, options$label, options$fig.cur, options$fig.ext, options$fig.cap)
+
+})
+
+opts_chunk$set(
+  echo=TRUE,
+  fig.path="static/geoknife-data/",
+  fig.width = 6,
+  fig.height = 6,
+  fig.cap = "TODO"
+)
+
+set.seed(1)
+```
+
+## Remote processing
+
+The USGS Geo Data Portal is designed to perform web-service processing on large gridded datasets. `geoknife` allows R users to take advantage of these services by processing large data, such as data available in the GDP catalog or a user-defined dataset, on the web. This type of workflow has three main advantages: 
+
+1. it allows the user to avoid downloading large datasets, 
+2. it avoids reinventing the wheel for the creation and optimization of complex geoprocessing algorithms, and 
+3. computing resources are dedicated elsewhere, so geoknife operations do not have much of an impact on a local computer.
+
+## geoknife components: fabric, stencil, knife
+
+The main components of a geoknife workflow are the fabric, stencil, and knife. These three components go into the final element, the geoknife "job" (geojob), which returns the processed data. The fabric is the gridded web dataset to be processed, the stencil is the feature of interest, and the knife is the processing algorithm parameters. Each of the `geoknife` components is created using a corresponding function: fabrics are created using `webdata()`, stencils are created using `webgeom()`, and knives are created using `webprocess()`. 
+
+This lesson will focus on discovering what options exist for each of those components. The next lesson will teach you how to construct each component and put it together to get the processed data. Before continuing this lesson, load the geoknife library (install if necessary).
+
+```{r load-geoknife}
+# load the geoknife package
+library(geoknife)
+```
+
+## Available webdata
+
+### GDP datasets
+
+To learn what data is available through GDP, you can explore the [online catalog](https://cida.usgs.gov/gdp/) or use the function `query`. Note that the access pattern for determining available fabrics differs slightly from stencils and knives. Running `query("webdata")` will return every dataset available through GDP:
+
+```{r load-webdata, message=FALSE, warning=FALSE}
+all_webdata <- query("webdata")
+head(all_webdata)
+length(all_webdata)
+```
+
+Notice that the object returned is a special `geoknife` class of `datagroup`. There are specific `geoknife` functions that only operate on an object of this class, see `?title` and `?abstract`. These two functions are used to extract metadata information about each of the available GDP datasets. With `r length(all_webdata)` datasets available, it is likely that reading through each to find ones that are of interest to you would be time consuming. You can use `grep` along with the accessor functions `title` and `abstract` to figure out which datasets you would like to use for processing. 
```

Comment:
It would also be incorrect because you can use those functions to change the values, too. :thinking: what if I just didn't name that category?